pkg/models/config.go

```diff
@@ -15,25 +15,32 @@ type Record struct {
 	PassThroughPorts []uint  `json:"passThroughPorts" yaml:"passThroughPorts"`
 	Filters          Filters `json:"filters" yaml:"filters"`
 }
+
 type Filters struct {
 	ReqHeader  []string            `json:"req_header" yaml:"req_header"`
 	URLMethods map[string][]string `json:"urlMethods" yaml:"urlMethods"`
 }
+
 type Test struct {
-	Path               string   `json:"path" yaml:"path"`
-	Command            string   `json:"command" yaml:"command"`
-	ProxyPort          uint32   `json:"proxyport" yaml:"proxyport"`
-	ContainerName      string   `json:"containerName" yaml:"containerName"`
-	NetworkName        string   `json:"networkName" yaml:"networkName"`
-	TestSets           []string `json:"testSets" yaml:"testSets"`
-	GlobalNoise        string   `json:"globalNoise" yaml:"globalNoise"`
-	Delay              uint64   `json:"delay" yaml:"delay"`
-	ApiTimeout         uint64   `json:"apiTimeout" yaml:"apiTimeout"`
-	PassThroughPorts   []uint   `json:"passThroughPorts" yaml:"passThroughPorts"`
-	WithCoverage       bool     `json:"withCoverage" yaml:"withCoverage"` // boolean to capture the coverage in test
+	Path             string        `json:"path" yaml:"path"`
+	Command          string        `json:"command" yaml:"command"`
+	ProxyPort        uint32        `json:"proxyport" yaml:"proxyport"`
+	ContainerName    string        `json:"containerName" yaml:"containerName"`
+	NetworkName      string        `json:"networkName" yaml:"networkName"`
+	TestSets         []string      `json:"testSets" yaml:"testSets"`
+	GlobalNoise      Globalnoise   `json:"globalNoise" yaml:"globalNoise"`
+	Delay            uint64        `json:"delay" yaml:"delay"`
+	ApiTimeout       uint64        `json:"apiTimeout" yaml:"apiTimeout"`
+	PassThroughPorts []uint        `json:"passThroughPorts" yaml:"passThroughPorts"`
+  WithCoverage       bool     `json:"withCoverage" yaml:"withCoverage"` // boolean to capture the coverage in test
 	CoverageReportPath string   `json:"coverageReportPath" yaml:"coverageReportPath"` // directory path to store the coverage files
 }
 
+type Globalnoise struct {
```

Comment:
Have you checked the case where I dont give Testsets noise at all..? like this :

```
globalNoise:
    global:
      body: {}
      header: {}
```

If I don't give test-sets noise at all will it throw an error while converting from yaml to go struct.

In the same way if I just include this without header json will it throw an error or ignore it.

```
globalNoise:
    global:
      body: {}
```

Reply:
I have tested this from my end it is working fine. Please resolve the other comment.